### PR TITLE
Display data on card for selected record

### PR DESCRIPTION
**Pull request by aboubakar20**
Screenshot_1

Screenshot_2

### CODE REVIEW

File: src/components/PostsTable.vue [hunk 1]
```diff
@@ -22,6 +22,8 @@ export default {
   },
   data() {
     return {
+      isShow: true,
+      postsData: [],
       columnDefs: this.columnDef(),
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: Can we change this function name to `getColumnDefs` ?

File: src/components/PostsTable.vue [hunk 2]
```diff
@@ -46,11 +48,13 @@ export default {
...
         {
           headerName: "Action",
-          field: "data.id",
+          field: "data",
           cellRenderer: "detailBtnCellRenderer",
           cellRendererParams: {
-            clicked: function(id) {
-              alert(`${id} was clicked`);
+            clicked: (postsData) => {
+              let data = this.postsData;
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: Can we avoid using `data` ?

File: src/App.vue [hunk 3]
```diff
@@ -2,21 +2,29 @@
...
 export default {
   name: "App",
   components: {
     PostsTable,
+    DetailCard,
   },
   data() {
     return {
       posts: null,
+      detailCardData: null,
```
Comment (junaidrt) [inline src/App.vue]: `detailCardData` -> `cardDetails` ?

File: src/App.vue [hunk 4]
```diff
@@ -30,6 +38,11 @@ export default {
         console.log(error);
```
Comment (junaidrt) [inline src/App.vue]: Please remove these console statements.

File: src/components/PostsTable.vue [hunk 5]
```diff
@@ -46,11 +48,13 @@ export default {
...
+              data.push(postsData);
+              this.$emit("postData", data, this.isShow);
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: this.$emit("postData", [postsData]);

File: src/App.vue [hunk 6]
```diff
@@ -30,6 +38,11 @@ export default {
...
       }
     },
+
+    getAndShowPostDetail(postDetail, isShow) {
+      this.detailCardData = postDetail;
+      this.isShow = isShow;
```
Comment (junaidrt) [inline src/App.vue]: `this.isShow = true`

**Review by junaidrt — CHANGES_REQUESTED** [review]

File: src/components/PostsTable.vue (hunk 1 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Changed

File: src/components/PostsTable.vue (hunk 2 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Removed

File: src/App.vue (hunk 3 above)
Comment (aboubakar20) [reply src/App.vue]: Changed

File: src/App.vue (hunk 6 above)
Comment (aboubakar20) [reply src/App.vue]: Done

File: src/components/PostsTable.vue (hunk 5 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Changed

File: src/App.vue (hunk 4 above)
Comment (aboubakar20) [reply src/App.vue]: Done

**Comment by aboubakar20** [comment]
Screenshot_3

File: src/components/PostsTable.vue [hunk 7]
```diff
@@ -46,11 +47,12 @@ export default {
         },
         {
           headerName: "Action",
-          field: "data.id",
+          field: "data",
           cellRenderer: "detailBtnCellRenderer",
           cellRendererParams: {
-            clicked: function(id) {
-              alert(`${id} was clicked`);
+            clicked: (posts) => {
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: `posts` -> `postDetail` or `selectedPost` ?

**Review by junaidrt — CHANGES_REQUESTED** [review]

File: src/components/PostsTable.vue (hunk 7 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Changed

**Review by junaidrt — APPROVED** [review]

**Pull request merged**